custom/enikshay/management/commands/run_diagnosis_test_migration.py

```diff
@@ -0,0 +1,108 @@
+from custom.enikshay.exceptions import ENikshayCaseNotFound
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument(
+            "log_path",
+            help="Path to write the log to"
+        )
+        parser.add_argument(
+            '--commit',
+            action='store_true',
+            help="Actually modifies the cases. Without this flag, it's a dry run."
+        )
+
+    def handle(self, domain, log_path, **options):
+        commit = options['commit']
+        accessor = CaseAccessors(domain)
+        factory = CaseFactory(domain)
+        headers = [
+            'case_id',
+            'diagnosis_test_result_date',
+            'diagnosis_lab_facility_name',
+            'diagnosis_test_lab_serial_number',
+            'diagnosis_test_summary',
+            'datamigration_diagnosis_test_information',
+        ]
+
+        print("Starting {} migration on {} at {}".format(
+            "real" if commit else "fake", domain, datetime.datetime.utcnow()
+        ))
+
+        with open(log_path, "w") as log_file:
+            writer = csv.writer(log_file)
+            writer.writerow(headers)
+
+            for episode_case_id in accessor.get_case_ids_in_domain(type='episode'):
+                print('Looking at {}'.format(episode_case_id))
+                episode_case = accessor.get_case(episode_case_id)
+                case_properties = episode_case.dynamic_case_properties()
+
+                if self.should_migrate_case(case_properties):
+                    test = self.get_relevant_test_case(domain, episode_case)
+
+                    if test is not None:
+                        update = self.get_updates(test)
+                        print('Updating {}...'.format(episode_case_id))
+                        writer.writerow([episode_case_id] + [update[key] for key in headers[1:]])
+
+                        if commit:
+                            factory.update_case(case_id=episode_case_id, update=update)
+                    else:
+                        print('No relevant test found for episode {}'.format(episode_case_id))
+                else:
+                    print('Do not migrate {}'.format(episode_case_id))
+
+        print('Migration complete at {}'.format(datetime.datetime.utcnow()))
+
+    @staticmethod
+    def should_migrate_case(case_properties):
+        return (
+            case_properties.get('datamigration_diagnosis_test_information') != 'yes'
+            and case_properties.get('episode_type') == 'confirmed_tb'
+            and case_properties.get('treatment_initiated') == 'yes_phi'
+            and case_properties.get('diagnosis_test_result_date', '') == ''
+        )
+
+    @staticmethod
+    def get_relevant_test_case(domain, episode_case):
+        try:
+            occurrence_case = get_occurrence_case_from_episode(domain, episode_case.case_id)
+        except ENikshayCaseNotFound:
+            return None
```

Comment:
@nickpell 
Is there a way to log these cases so we know about them?

Reply:
done